sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -240,125 +212,30 @@
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ConfigurationSetting>> setSettingWithResponse(ConfigurationSetting setting) {
-        return withContext(context -> setSetting(setting, context));
+    public Mono<Response<ConfigurationSetting>> setSettingWithResponse(ConfigurationSetting setting,
+                                                                       boolean onlyIfUnchanged) {
+        return withContext(context -> setSetting(setting, onlyIfUnchanged, context));
     }
 
-    Mono<Response<ConfigurationSetting>> setSetting(ConfigurationSetting setting, Context context) {
+    Mono<Response<ConfigurationSetting>> setSetting(ConfigurationSetting setting, boolean onlyIfUnchanged,
+                                                    Context context) {
         // Validate that setting and key is not null. The key is used in the service URL so it cannot be null.
         validateSetting(setting);
 
+        final String ifMatchETag = onlyIfUnchanged ? getETagValue(setting.getETag()) : "*";
```

Comment:
I don't think we should pass an `ETag` if there is no check for unchanged, this will cause the set to fail if the setting doesn't exist.